include/lsst/afw/cameraGeom/Detector.h

```diff
@@ -300,35 +259,391 @@ class Detector final : public table::io::PersistableFacade<Detector>, public typ
+             std::vector<std::shared_ptr<Amplifier const>> && amplifiers);
 
     std::string getPersistenceName() const override;
 
     std::string getPythonModule() const override;
 
     void write(OutputArchiveHandle& handle) const override;
 
-    std::string _name;                      ///< name of detector's location in the camera
-    int _id;                                ///< detector numeric ID
-    DetectorType _type;                     ///< type of detectorsize_t
-    std::string _serial;                    ///< serial "number" that identifies the physical detector
-    lsst::geom::Box2I _bbox;                ///< bounding box
-    std::vector<std::shared_ptr<Amplifier const>> _amplifiers;  ///< list of amplifiers
-    _AmplifierMap _amplifierMap;            ///< map of amplifier name: amplifier
-    Orientation _orientation;               ///< position and orientation of detector in focal plane
-    lsst::geom::Extent2D _pixelSize;        ///< pixel size (mm)
-    CameraSys _nativeSys;                   ///< native coordinate system of this detector
-    std::shared_ptr<TransformMap const> _transformMap;   ///< registry of coordinate transforms
-    CrosstalkMatrix _crosstalk;             ///< crosstalk coefficients
-    std::string _physicalType;              ///< the physical type of the detector (e.g. CCD, ITL, E2V)
+    Fields const _fields;
+    std::shared_ptr<TransformMap const> _transformMap;
+    std::vector<std::shared_ptr<Amplifier const>> _amplifiers;
 };
+
+
+/**
+ * A helper class for Detector that allows amplifiers and most fields to be
+ * modified.
+ *
+ * Because Detector is immutable, creation and modification always go through
+ * Builder, or more precisely, one of its two subclasses:
+ *  - InCameraBuilder (obtained from Camera::Builder) is used to construct
+ *    new detectors or modify them as part of Camera.
+ *  - PartialRebuilder (obtained from Detector::rebuild) is used to modify
+ *    an existing detector without changing its relationship to its camera.
+ *
+ * Detector::Builder itself provides functionality common to these:
+ *  - setters for the simple data fields of Detector;
+ *  - a container of Amplifier::Builders.
+ *
+ * The name and ID of a detector (but not its "serial" string) are set at
+ * initial construction and are an integral part of the relationship between
+ * it and its Camera, and can never be changed, even by Builders.
+ *
+ * The fact that Amplifier::Builder inherits from Amplifier does not mean that
+ * a container of Amplifier::Builder can inherit from a container of Amplifier,
+ * and hence Detector::Builder cannot inherit directly from Detector.  But in
+ * both Python and templated C++ code, the container interfaces of Detector
+ * and Detector::Builder are identical (i.e. they're "duck type" equivalent),
+ * aside from the fact that Detector::Builder also permits addition and removal
+ * of amplifiers.
```

Comment:
Maybe change "and hence Detector::Builder cannot inherit directly from Detector" to "and hence (because of implementation details) Detector::Builder cannot inherit directly from Detector", to make it clearer? I'm not used to inherited state being an issue, and others might not be, either.